server/src/discord/discord.service.ts

```diff
@@ -0,0 +1,67 @@
+import { ForbiddenException, Injectable, Res } from '@nestjs/common';
+import { Request } from '@nestjs/common';
+import axios from 'axios';
+import { PrismaService } from '../prisma/prisma.service';
+import { ServiceType } from '@prisma/client';
+
+@Injectable()
+export class DiscordService {
+  constructor(private prisma: PrismaService) {}
+  async discordLogin(@Res() req: any) {
+    const url =
+      'https://discord.com/api/oauth2/authorize?client_id=1189230397108793416&response_type=code&redirect_uri=http%3A%2F%2Flocalhost%3A8080%2Fauth%2Fdiscord%2Fcallback&scope=identify+email';
+    req.redirect(url);
```

Comment:
Here we need a strategy caus the client id is exposed

Reply:
changed